src/core/packages/elasticsearch/server-internal/src/elasticsearch_service.test.ts

```diff
@@ -224,41 +227,94 @@ describe('#setup', () => {
 
   it('esNodeVersionCompatibility$ only starts polling when subscribed to', async () => {
     const mockedClient = mockClusterClientInstance.asInternalUser;
-    mockedClient.nodes.info.mockImplementation(() =>
-      elasticsearchClientMock.createErrorTransportRequestPromise(new Error())
-    );
+    // With retry: 1, each poll makes 2 attempts (initial + 1 retry)
+    // Mock 4 total error responses: 2 for first poll + 2 for second poll
+    mockedClient.nodes.info
+      .mockImplementationOnce(() =>
+        elasticsearchClientMock.createErrorTransportRequestPromise(new Error())
+      )
+      .mockImplementationOnce(() =>
+        elasticsearchClientMock.createErrorTransportRequestPromise(new Error())
+      )
+      .mockImplementationOnce(() =>
+        elasticsearchClientMock.createErrorTransportRequestPromise(new Error())
+      )
+      .mockImplementationOnce(() =>
+        elasticsearchClientMock.createErrorTransportRequestPromise(new Error())
+      );
 
     expect(mockedClient.nodes.info).toHaveBeenCalledTimes(0);
 
     const setupContract = await elasticsearchService.setup(setupDeps);
 
+    // Initial call should happen synchronously
     expect(mockedClient.nodes.info).toHaveBeenCalledTimes(1);
 
-    tick();
+    // Advance timers to trigger interval emission (this allows retry delay to emit)
+    await jest.advanceTimersByTimeAsync(TICK);
 
+    // After retry completes (triggered by interval emission)
     expect(mockedClient.nodes.info).toHaveBeenCalledTimes(2);
 
+    // Wait for the observable to emit (it should have cached value from shareReplay)
```

Comment:
The comments might be overkill. I kept them in to help explain the what's happening